authors/apps/comments/views.py

```diff
@@ -5,37 +5,59 @@
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from rest_framework import generics
-
+from .utils import validationData
 from authors.apps.articles.models import Article
 from authors.apps.profiles.models import Profile
 from .serializers import (CommentSerializer,
                             CommentLikesDislikeSerializer)
 from .models import (Comment, CommentLikeDislike)
 from .permissions import IsOwnerOrReadOnly
 
-
 class ListCreateCommentView(APIView):
     permission_classes = [IsAuthenticatedOrReadOnly]
     serializer_class = CommentSerializer
 
     def post(self, request, *args, **kwargs):
-        comment = request.data
+        comment = request.data.get('comment', {})
 
         # get the article or return a 404 error
         article = get_object_or_404(Article, slug=kwargs["slug"])
 
         # user the current Id pf user to get their Profile object
         current_user = request.user
         author = Profile.objects.get(user=current_user)
+
+        end_index = comment.get('end_index', 0)
+        start_index = comment.get('start_index', 0)
+        if "start_index" in comment and "end_index" in comment:
+            check = validationData(end_index,start_index,article)
+            if check == True:
```

Comment:
check is not a very meaningful variable name. You could try [is_data_valid]

Reply:
Okay let me change that